fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/source/PaimonScanNode.java

```diff
@@ -295,33 +331,21 @@ public List<Split> getSplits(int numBackends) throws UserException {
             splitStats.add(splitStat);
         }
 
+        // if applyCountPushdown is true, calcute row count for count pushdown
+        if (applyCountPushdown) {
+            Optional<Long> optTableLevelCount = calcuteTableLevelCount(paimonSplits);
+            if (optTableLevelCount.isPresent()) {
+                long tableLevelRowCount = optTableLevelCount.get();
+                setPushDownCount(tableLevelRowCount);
+                assignCountToSplits(splits, tableLevelRowCount);
```

Comment:
We can make an optimization here. If there are too many splits, we can only select a small number of splits. For specific details, you can refer to Iceberg.